packages/path-provider/test/path_provider_test.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:flutter/services.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test('Path provider control test', () async {
+    final List<MethodCall> log = <MethodCall>[];
+    String response;
+    const channel = const MethodChannel('plugins.flutter.io/path_provider');
+
+    channel.setMockMethodCallHandler((MethodCall methodCall) async {
+      log.add(methodCall);
+      return response;
+    });
+
+    Directory directory = await getTemporaryDirectory();
+
+    expect(log, equals(<MethodCall>[new MethodCall('getTemporaryDirectory')]));
+    expect(directory, isNull);
+    log.clear();
+
```

Comment:
We start a new test here. Should ideally be separated into two invocations of `test`.